openlibrary/plugins/upstream/addbook.py

```diff
@@ -639,13 +637,25 @@ def process_work(self, work):
         :rtype: web.storage
         """
         def read_subject(subjects):
+            """
+            >>> list(read_subject("A,B,C,B")) == [u'A', u'B', u'C']   # str
+            True
+            >>> list(read_subject(r"A,B,C,B")) == [u'A', u'B', u'C']  # raw
+            True
+            >>> list(read_subject(u"A,B,C,B")) == [u'A', u'B', u'C']  # Unicode
+            True
+            >>> list(read_subject(""))
+            []
+            """
             if not subjects:
                 return
-
-            f = StringIO(subjects.encode('utf-8'))  # no unicode in csv module
+            if six.PY2:
+                subjects = subjects.encode('utf-8')  # no unicode in csv module
+            f = six.StringIO(subjects)
             dedup = set()
             for s in next(csv.reader(f, dialect='excel', skipinitialspace=True)):
-                s = s.decode('utf-8')
+                if six.PY2:
+                    s = s.decode('utf-8')
```

Comment:
Were these meant to be part of this PR? if they're fixing Py2/3 bugs, let's pull them into a separate PR so we can test to makes sure they work on dev.openlibrary.org with Py2.

Reply:
DONE